README.md

```diff
@@ -1,2 +1,250 @@
+
+A bean class has the following constraints:
+
+- if a class has multiple constructors, mark the one with the fields that should be injected by Bring with
+  [@Inject](src/main/java/com/bobocode/hoverla/bring/annotation/Inject.java) annotation. If a class has only one
+  constructor, the annotation is not necessary
+- use only [constructor injection](#constructor-injection) or [field injection](#field-injection).
+  Using of both injection types **is not recommended**, because it may lead to unpredictable results
+
+### Java configuration
+
+---
+With Java Configuration you can provide a way how a bean should be instantiated via Java code. Use
+a [@Configuration](src/main/java/com/bobocode/hoverla/bring/annotation/Configuration.java)
+annotation to mark a class that will contain bean declarations. To declare a bean create a method and mark it
+as [@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java)
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+import com.bobocode.hoverla.bring.annotation.Configuration;
+
+@Configuration
+public class MyJavaConfiguration {
+
+    @Bean
+    public MyBean myBean() {
+        return new MyBean();
+    }
+}
+```
+
+By default, **Bring will use a method name as a bean name**. The bean name can be provided via
+the [@Bean](src/main/java/com/bobocode/hoverla/bring/annotation/Bean.java) annotation
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+
+public class MyJavaConfiguration {
+
+    @Bean(name = "customBean")
+    public MyCustomBean myCustomBean() {
+        return new MyCustomBean();
+    }
+}
+```
+
+If you want other beans to be injected inside a method, provide them as parameters
+
+```java
+import com.bobocode.hoverla.bring.annotation.Configuration;
+
+@Configuration
+public class MyJavaConfiguration {
+
+    @Bean
+    public MyBean myBean(InjectableBean injectableBean) {
+        return new MyBean(injectableBean);
+    }
+}
+```
+
+To specify a bean name use a [@Qualifier](src/main/java/com/bobocode/hoverla/bring/annotation/Qualifier.java) annotation
+
+```java
+import com.bobocode.hoverla.bring.annotation.Configuration;
+
+@Configuration
+public class MyJavaConfiguration {
+
+    @Bean
+    public MyBean myBean(InjectableBean injectableBean, @Qualifier("customBean") MyCustomBean myCustomBean) {
+        return new MyBean(injectableBean, myCustomBean);
+    }
+}
+```
+
+### Constructor injection
+
+---
+To use constructor injection, provide beans, that you want to be injected, as parameters to your bean's constructor.
+If the bean has multiple constructors, the
+annotation [@Inject](src/main/java/com/bobocode/hoverla/bring/annotation/Inject.java)
+should be provided. If only one constructor is present, the annotation is not necessary
+
+```java
+import com.bobocode.hoverla.bring.annotation.Bean;
+import com.bobocode.hoverla.bring.annotation.Inject;
+
+@Bean
+public class MyBean {
+
+    private final InjectableBean injectableBean;
```

Comment:
It will not be compiled if you leave 'final' here.  Please try to show working examples of code in the Readme file to give users opportunity to copy and paste this code for fast "hello world" application starting

Reply:
fixed it